src/test/java/com/yelp/nrtsearch/server/luceneserver/search/query/ConstantScoreQueryTest.java

```diff
@@ -0,0 +1,172 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class ConstantScoreQueryTest extends ServerTestCase {
+  @ClassRule public static final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  protected List<String> getIndices() {
+    return Collections.singletonList(DEFAULT_TEST_INDEX);
+  }
+
+  protected FieldDefRequest getIndexDef(String name) throws IOException {
+    return getFieldsFromResourceFile("/search/query/registerFieldsConstScore.json");
+  }
+
+  protected void initIndex(String name) throws Exception {
+    List<String> textValues = List.of("t1 t2 t3", "t1 t3", "t4 t5 t6", "t2 t6 t7", "t1 t2 t8");
+    List<AddDocumentRequest> docs = new ArrayList<>();
+    int index = 0;
+    for (String textValue : textValues) {
+      AddDocumentRequest request =
+          AddDocumentRequest.newBuilder()
+              .setIndexName(name)
+              .putFields(
+                  "doc_id", MultiValuedField.newBuilder().addValue(Integer.toString(index)).build())
+              .putFields("text_field", MultiValuedField.newBuilder().addValue(textValue).build())
+              .build();
+      docs.add(request);
+      index++;
+    }
+    addDocuments(docs.stream());
+  }
+
+  @Test
+  public void testWithoutConstantScoreQuery() {
+    SearchResponse response =
+        getGrpcServer()
+            .getBlockingStub()
+            .search(
+                SearchRequest.newBuilder()
+                    .setIndexName(DEFAULT_TEST_INDEX)
+                    .setTopHits(10)
+                    .addRetrieveFields("doc_id")
+                    .setQuery(
+                        Query.newBuilder()
+                            .setMatchQuery(
+                                MatchQuery.newBuilder()
+                                    .setField("text_field")
+                                    .setQuery("t2")
+                                    .build())
+                            .build())
+                    .build());
+    assertIds(response, 0, 3, 4);
+    for (Hit hit : response.getHitsList()) {
+      assertNotEquals(1.0, hit.getScore());
+    }
+  }
+
+  @Test
+  public void testConstantScoreQuery() {
+    SearchResponse response =
+        getGrpcServer()
+            .getBlockingStub()
+            .search(
+                SearchRequest.newBuilder()
+                    .setIndexName(DEFAULT_TEST_INDEX)
+                    .setTopHits(10)
+                    .addRetrieveFields("doc_id")
+                    .setQuery(
+                        Query.newBuilder()
+                            .setConstantScoreQuery(
+                                ConstantScoreQuery.newBuilder()
+                                    .setFilter(
+                                        Query.newBuilder()
+                                            .setMatchQuery(
```

Comment:
Is it possible to add another match query to see if the final calculated score is the sum of the scores or the multiplication of them?

Reply:
The score of the match query is not used, as it is replaced by the constant score.